cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -127,34 +142,75 @@ def pad_index_two(index_string: str, reverse_complement: bool) -> str:
     return index_string + INDEX_TWO_PAD_SEQUENCE
 
 
-def adapt_indexes(
-    samples: List[FlowCellSample],
-    run_parameters: RunParameters,
+def get_hamming_distance_for_indexes(sequence_1: str, sequence_2: str) -> int:
+    """Get the hamming distance between two index sequences.
+    In the case that one sequence is longer than the other, the distance is calculated between
+    the shortest sequence and the first segment of equal length of the longest sequence."""
+    limit: int = min(len(sequence_1), len(sequence_2))
+    return get_hamming_distance(str_1=sequence_1[:limit], str_2=sequence_2[:limit])
+
+
+def update_barcode_mismatch_values_for_sample(
+    sample_to_update: FlowCellSampleNovaSeqX, samples: List[FlowCellSampleNovaSeqX]
```

Comment:
Can we have some more descriptive variable names here? `sample` is often the more important variable in a function, but here one wants to focus on sample_to_update
```suggestion
    sample_to_update: FlowCellSampleNovaSeqX, samples_to_compare_to: List[FlowCellSampleNovaSeqX]
```